modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/QueryMemoryManager.java

```diff
@@ -190,29 +182,44 @@ public QueryMemoryManager(GridKernalContext ctx) {
      * Note: If 'maxQueryMemory' is zero, then {@link QueryMemoryManager#qryQuota}  will be used.
      * Note: Negative values are reserved for disable memory tracking.
      *
-     * @param maxQueryMemory Query memory limit in bytes.
+     * @param maxQryMemory Query memory limit in bytes.
      * @return Query memory tracker.
      */
-    public QueryMemoryTracker createQueryMemoryTracker(long maxQueryMemory) {
-        assert maxQueryMemory >= 0;
+    public QueryMemoryTracker createQueryMemoryTracker(long maxQryMemory, IgniteTrace trace) {
+        assert maxQryMemory >= 0;
 
-        if (maxQueryMemory == 0)
-            maxQueryMemory = qryQuota;
+        if (maxQryMemory == 0)
+            maxQryMemory = qryQuota;
 
         long globalQuota0 = globalQuota;
 
-        if (maxQueryMemory == 0 && globalQuota0 == 0)
-            return null; // No memory tracking configured.
+        if (maxQryMemory == 0 && globalQuota0 == 0) {
+            if (log.isDebugEnabled()) {
+                log.debug("No memory quota configured for the query. " +
+                    "It will be executed without memory tracking: "  + trace.queryDescriptor());
+            }
 
-        if (globalQuota0 > 0 && globalQuota0 < maxQueryMemory) {
-            U.warn(log, "Max query memory can't exceed SQL memory pool size. Will be reduced down to: " + globalQuota);
+            return null;
+        }
+
+        if (globalQuota0 > 0 && globalQuota0 < maxQryMemory) {
+            if (log.isDebugEnabled()) {
+                log.debug( "Max query memory can't exceed SQL memory pool size. Will be reduced down to: " +
```

Comment:
WDYT, may be better to have `info` level for the case?

Reply:
Also potentially will be useful  to print the same warning for `setQueryQuota` method.

Reply:
I set info level with throttling and also added a warning on quota set up.